solr/core/src/java/org/apache/solr/filestore/DistribPackageStore.java

```diff
@@ -94,8 +94,18 @@ private static Path _getRealPath(String path, Path solrHome) {
     if (!path.isEmpty() && path.charAt(0) != File.separatorChar) {
       path = File.separator + path;
     }
-    // Use concat because path might start with a slash and be incorrectly interpreted as absolute
-    return solrHome.resolve(PackageStoreAPI.PACKAGESTORE_DIRECTORY + path);
+    if (path.startsWith("\\\\")) { // Windows absolute UNC
+      throw new SolrException(BAD_REQUEST, "Illegal path " + path);
+    }
+    while (path.startsWith("/")) { // Trim all leading slashes
```

Comment:
This is odd because at line 94 we just added leading slashes?

Reply:
Yes, and the reason the old code added a leading slash was that it id string concatenation `PackageStoreAPI.PACKAGESTORE_DIRECTORY + path`, while the new code does `resolve(path)`. With string concat you need a leading slash to get a sub folder of pkgStore, while if you have a leading slash with Path.resolve, you end up with a an absolute path. However I'll re-factor this whole part and add some tests.

Reply:
I specifically remember getting bit by this in ab6e6959

Reply:
Pushed an improvement

Reply:
> I specifically remember getting bit by this in [ab6e695](https://github.com/apache/solr/commit/ab6e695973ca8cc4c8161ef9ec07d3749bf0870f)

Yes, the Path API is a bit sneaky there. I'd ideally convert all those String representations of filesystem paths in the pkgMgr into Paths, but that's out of scope for this.

The test code uses strings with leading slashes as meaning "root of the file store", so that's why I strip all leading slashes (path-sep) so that the resolve is always relative, and in addition double check that we are not outside the filestore root.